esmvaltool/cmorizers/data/formatters/datasets/esacci_soilmoisture.py

```diff
@@ -0,0 +1,165 @@
+from cf_units import Unit
+
+from ...utilities import (
+    convert_timeunits,
+    fix_var_metadata,
+    fix_dim_coordnames,
+    fix_bounds,
+    save_variable,
+    set_global_atts,
+    roll_cube_data
+)
+
+logger = logging.getLogger(__name__)
+
+
+def fix_coords(cube):
+    """Fix coordinates to CMOR standards.
+
+    Fixes coordinates eg time to have correct units, bounds etc;
+    longitude to be CMOR-compliant 0-360deg; fixes some attributes
+    and bounds - the user can avert bounds fixing by using supplied
+    arguments; if bounds are None they will be fixed regardless.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        data cube with coordinates to be fixed.
+
+
+    Returns
+    -------
+    cube: iris.cube.Cube
+        data cube with fixed coordinates.
+    """
+    # first fix any completely missing coord var names
+    fix_dim_coordnames(cube)
+    # fix individual coords
+    for cube_coord in cube.coords():
+        # fix time
+        if cube_coord.var_name == 'time':
+            logger.info("Fixing time...")
+            cube.coord('time').convert_units(
+                Unit('days since 1970-01-01T00:00:00+00:00',
+                     calendar='proleptic_gregorian'))
+
+        # fix longitude
+        if cube_coord.var_name == 'lon':
+            logger.info("Fixing longitude...")
+            if cube_coord.ndim == 1:
+                if cube_coord.points[0] < 0. and \
+                        cube_coord.points[-1] < 181.:
+                    cube_coord.points = \
+                        cube_coord.points + 180.
+                    cube.attributes['geospatial_lon_min'] = 0.
+                    cube.attributes['geospatial_lon_max'] = 360.
+                    nlon = len(cube_coord.points)
+                    roll_cube_data(cube, nlon // 2, -1)
+        fix_bounds(cube, cube_coord)
+
+    return cube
+
+
+def extract_variable(var_info, raw_info, year):
+    """Extract variables."""
+    rawvar = raw_info['name']
+    constraint = iris.Constraint(name=rawvar)
+    if rawvar == 'sm_uncertainty':
+        sm_cube = iris.load_cube(raw_info['file'],
+                                 iris.NameConstraint(var_name='sm'))
+        ancillary_var = sm_cube.ancillary_variable(
+            'Volumetric Soil Moisture Uncertainty'
+        )
+        cube = sm_cube.copy(ancillary_var.core_data())
+    else:
+        cube = iris.load_cube(raw_info['file'], constraint)
+
+    # Continue with processing the cube if it was successfully loaded
+    fix_var_metadata(cube, var_info)
+    convert_timeunits(cube, year)
+    fix_coords(cube)
+
+    # Remove dysfunctional ancillary data without standard names
+    for ancillary_variable in cube.ancillary_variables():
+        cube.remove_ancillary_variable(ancillary_variable)
+
+    return cube
+
+
+def cmorization(in_dir, out_dir, cfg, cfg_user, start_date, end_date):
+    """Cmorize data."""
+    glob_attrs = cfg['attributes']
```

Comment:
```suggestion
    glob_attrs = cfg['attributes']
    if not start_date:
        start_date = datetime(1978, 1, 1)
    if not end_date:
        end_date = datetime(2022, 12, 31)
```

Reply:
Apparently in CEDA the time period for this version starts from 01/11/1978 to 31/12/2022, but even if I leave the start_date 01/01/1978 the cmorised output files start from 01/11/1978 based on the data availability.